x-pack/solutions/chat/plugins/wci-index-source/public/hooks/use_index_name_autocomplete.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useQuery } from '@tanstack/react-query';
+import type { CoreStart } from '@kbn/core/public';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import type { SearchIndicesResponse } from '../../common/http_api/configuration';
+
+export const useIndexNameAutocomplete = () => {
+  const {
+    services: { http, notifications },
+  } = useKibana<CoreStart>();
+
+  const { isLoading, data } = useQuery({
+    queryKey: ['index-name-autocomplete'],
+    queryFn:
+      () =>
+      async ({ indexName }: { indexName: string }) => {
+        const response = await http.get<SearchIndicesResponse>(
+          `/internal/wci-index-source/indices-autocomplete`,
+          {
+            query: {
+              index: indexName,
+            },
+          }
+        );
+        return response.indexNames;
+      },
+    onError: (err: any) => {
+      notifications.toasts.addError(err, { title: 'Error fetching indices' });
```

Comment:
If we are surfacing the error in a toast, we should wrap the error string with the i18n logic

Reply:
I've copied it over from another WorkChat component - none of components are using i18n, so it should be fine to not do it here? Or should I change this everywhere in the code within the PR?

Reply:
i think it should be fine to not worry about i18n at this stage